cmd/oras/root/push.go

```diff
@@ -139,6 +141,8 @@ func runPush(ctx context.Context, opts pushOptions) error {
 		return err
 	}
 	defer store.Close()
+	memStore := memory.New()
+	hybrid := ostore.NewReadOnlyHybrid(store, memStore, memStore)
```

Comment:
Based on other comments, you may consider
```suggestion
	union := contentutil.MultiReadOnlyTarget(memStore, Store)
```